### PR TITLE
New steering

### PR DESCRIPTION
**Pull request by ArchDemons**
added faceing creatures
rollback some debug stuff
fixed bug with tile tooltip
fixed creature steering

### CODE REVIEW

**Comment by tonihele** [comment]
Heh, rather comical. The speed is very high now, they miss their targets quite often and spin wildly near target. But otherwise the facing is nice. I think it needs some fixes before merging.

**Comment by ArchDemons** [comment]
we can slowing rotation. But looks that bad. Something wrong with calculation angular acceleration or not limits it. Or we need firstly check acceleration first, than apply them...

**Comment by ArchDemons** [comment]
Hmmm, this looks different on my other computer... How it may happen?

File: src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java [hunk 1]
```diff
@@ -105,19 +102,27 @@ protected void applySteering(SteeringAcceleration<Vector2> steering, float tpf)
...
+            return;
         }
 
         // Update orientation and angular velocity
-        if (independentFacing) {
-            setOrientation(getOrientation() + (angularVelocity * tpf));
-            angularVelocity += steering.angular * tpf;
+        if (!stopMoving) {
+            position.add(linearVelocity.x * tpf, linearVelocity.y * tpf);
+            linearVelocity.mulAdd(steering.linear, tpf).limit(maxLinearSpeed);
```
Comment (ufdada) [inline src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java]: Are these 2 lines are really necessary? They are getting called some lines above, so if it runs into this block it is called twice.

At least my tests made them move like before (without the weird spinning)

File: src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java [hunk 2]
```diff
@@ -37,23 +38,27 @@
...
+    private float speedThreshold = 0.1f;
     private volatile boolean applySteering = false;
+    // FIXME: dirty hack. Roll back changes and see what happens :-(
+    private volatile boolean stopMoving = false;
 
     public AbstractCreatureSteeringControl(Creature creature) {
         this.creature = creature;
 
         maxLinearSpeed = creature.getSpeed();
+        maxLinearSpeed = maxLinearSpeed * 10;
```
Comment (ArchDemons) [inline src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java]: `maxLinearAcceleration = maxLinearSpeed * 10;`

but that fix not fixed all bugs :(

Comment (ufdada) [inline src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java]: Look at the other reply ;)

File: src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java (hunk 1 above)
Comment (ArchDemons) [inline src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java]: Main idea:
-add FollowPath and Face steering
- creature move to target
- when moving stops (end point is reached), creature rotates around Y-axis (faced).

But on my other computer with downloaded gdx-ai library it feel different with other computer.

File: src/toniarts/openkeeper/world/creature/CreatureControl.java [hunk 3]
```diff
@@ -538,22 +549,42 @@ public short getOwnerId() {
...
-            }
+            // Debug
+//            if (Main.isDebug()) {
+//                worldState.getApp().enqueue(() -> {
+//                    worldState.drawPath(waypoints);
+//                });
+//            }
+
+            FollowPath<Vector2, LinePathParam> followPath = new FollowPath(this, new LinePath<>(getWaypoints(outPath), true), 2);
+            followPath.setDecelerationRadius(0.3f);
```
Comment (ArchDemons) [inline src/toniarts/openkeeper/world/creature/CreatureControl.java]: Seems like library ignoring this attribute. Or I wrong understood it.

Comment (ArchDemons) [inline src/toniarts/openkeeper/world/creature/CreatureControl.java]: imps steel cut corners when moves. https://gamedevelopment.tutsplus.com/tutorials/understanding-steering-behaviors-path-following--gamedev-8769 Smoothing the Movement header. Where this attribute in gdx-ai ?

**Comment by tonihele** [comment]
Sorry it took this long, it definitely looks better now. They still easily get stuck though. Like they don't know that they're at the destination already. They just keep the running animation and stay stationary forever. Something with the facing perhaps, since the tasks were digging & claiming a wall when I saw them do this.

The steering behavior parameters can be a bit of pain. I don't have much time on the weekend, but I can also try to debug it. It is really delicate fine tuning perhaps.

**Comment by tonihele** [comment]
This is _impossible_ :( I tried cherry picking the facing, just the facing. And I added collision avoidance system. They all work.. to a degree... It is coming painfully obvious that the system needs to be customized more (See: https://github.com/jsjolund/GdxDemo3D/blob/master/core/src/com/mygdx/game/steerers/FollowPathSteerer.java). I disabled them for now. Changing any parameter yields too random results. We need better understanding.

We also need a bit better hierarchy, more controlled structure for clarity. Proper MVC. As the code keeps growing. Inheriting the AbstractCreatureSteeringControl is not perhaps the best design. Somewhat related matter to this steering.

**Comment by ufdada** [comment]
Superseded by #269

**Pull request closed without merging**